kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisher.java

```diff
@@ -53,38 +49,40 @@
  * <p>
  * This terminal does not dispatch Events internally, as it relies on each event processor to listen to it's own Kafka
  * Topic.
+ * </p>
  *
  * @param <K> a generic type for the key of the {@link ProducerFactory}, {@link Producer} and
  *            {@link KafkaMessageConverter}
  * @param <V> a generic type for the value of the {@link ProducerFactory}, {@link Producer} and
  *            {@link KafkaMessageConverter}
+ *
  * @author Nakul Mishra
+ * @author Simon Zambrovski
+ * @author Lars Bilger
  * @since 3.0
  */
 public class KafkaPublisher<K, V> {
 
     private static final Logger logger = LoggerFactory.getLogger(KafkaPublisher.class);
 
-    private final SubscribableMessageSource<EventMessage<?>> messageSource;
+    private static final String EXCEPTION_KEY = KafkaPublisher.class.getName() + ".EXCEPTION";
+
     private final ProducerFactory<K, V> producerFactory;
     private final KafkaMessageConverter<K, V> messageConverter;
     private final MessageMonitor<? super EventMessage<?>> messageMonitor;
     private final String topic;
     private final long publisherAckTimeout;
 
-    private Registration eventBusRegistration;
-
     /**
      * Instantiate a {@link KafkaPublisher} based on the fields contained in the {@link Builder}.
      * <p>
-     * Will assert that the {@link SubscribableMessageSource} and {@link ProducerFactory} are not {@code null}, and will
+     * Will assert that {@link ProducerFactory} are not {@code null}, and will
      * throw an {@link AxonConfigurationException} if any of them is {@code null}.
      *
      * @param builder the {@link Builder} used to instantiate a {@link KafkaPublisher} instance
      */
-    protected KafkaPublisher(Builder<K, V> builder) {
+    private KafkaPublisher(Builder<K, V> builder) {
```

Comment:
The constructor is intended to be `protected`, so that users can extend from the `KafkaPublisher` and it's `Builder` to add functionality, if desired.